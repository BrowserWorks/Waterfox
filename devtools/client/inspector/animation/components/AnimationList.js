/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const AnimationItem = createFactory(
  require("devtools/client/inspector/animation/components/AnimationItem")
);

class AnimationList extends PureComponent {
  static get propTypes() {
    return {
      animations: PropTypes.arrayOf(PropTypes.object).isRequired,
      emitEventForTest: PropTypes.func.isRequired,
      getAnimatedPropertyMap: PropTypes.func.isRequired,
      getNodeFromActor: PropTypes.func.isRequired,
      onHideBoxModelHighlighter: PropTypes.func.isRequired,
      onShowBoxModelHighlighterForNode: PropTypes.func.isRequired,
      selectAnimation: PropTypes.func.isRequired,
      setHighlightedNode: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      simulateAnimation: PropTypes.func.isRequired,
      timeScale: PropTypes.object.isRequired,
    };
  }

  render() {
    const {
      animations,
      emitEventForTest,
      getAnimatedPropertyMap,
      getNodeFromActor,
      onHideBoxModelHighlighter,
      onShowBoxModelHighlighterForNode,
      selectAnimation,
      setHighlightedNode,
      setSelectedNode,
      simulateAnimation,
      timeScale,
    } = this.props;

    return dom.ul(
      {
        className: "animation-list",
      },
      animations.map(animation =>
        AnimationItem({
          animation,
          emitEventForTest,
          getAnimatedPropertyMap,
          getNodeFromActor,
          onHideBoxModelHighlighter,
          onShowBoxModelHighlighterForNode,
          selectAnimation,
          setHighlightedNode,
          setSelectedNode,
          simulateAnimation,
          timeScale,
        })
      )
    );
  }
}

module.exports = AnimationList;
