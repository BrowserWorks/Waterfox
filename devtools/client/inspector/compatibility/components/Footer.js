/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const { PureComponent } = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const {
  updateSettingsVisibility,
} = require("devtools/client/inspector/compatibility/actions/compatibility");

loader.lazyRequireGetter(
  this,
  "openDocLink",
  "devtools/client/shared/link",
  true
);

const FEEDBACK_LINK =
  "https://docs.google.com/forms/d/e/1FAIpQLSeevOHveQ1tDuKYY5Fxyb3vvbKKumdLWUT5-RuwJWoAAOST5g/viewform";

const REPORT_ICON = "chrome://devtools/skin/images/report.svg";
const SETTINGS_ICON = "chrome://devtools/skin/images/settings.svg";

class Footer extends PureComponent {
  static get propTypes() {
    return {
      updateSettingsVisibility: PropTypes.func.isRequired,
    };
  }

  _renderButton(icon, label, onClick) {
    return dom.button(
      {
        className: "compatibility-footer__button",
        title: label,
        onClick,
      },
      dom.img({
        className: "compatibility-footer__icon",
        src: icon,
      }),
      dom.label(
        {
          className: "compatibility-footer__label",
        },
        label
      )
    );
  }

  render() {
    return dom.footer(
      {
        className: "compatibility-footer",
      },
      this._renderButton(
        SETTINGS_ICON,
        "Settings",
        this.props.updateSettingsVisibility
      ),
      this._renderButton(REPORT_ICON, "Feedback", () =>
        openDocLink(FEEDBACK_LINK)
      )
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateSettingsVisibility: () => dispatch(updateSettingsVisibility(true)),
  };
};

module.exports = connect(null, mapDispatchToProps)(Footer);
