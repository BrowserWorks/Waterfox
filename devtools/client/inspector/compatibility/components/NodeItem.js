/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { PureComponent } = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const {
  translateNodeFrontToGrip,
} = require("devtools/client/inspector/shared/utils");
const { REPS, MODE } = require("devtools/client/shared/components/reps/reps");
const { Rep } = REPS;
const ElementNode = REPS.ElementNode;

const Types = require("devtools/client/inspector/compatibility/types");

class NodeItem extends PureComponent {
  static get propTypes() {
    return {
      node: Types.node.isRequired,
      hideBoxModelHighlighter: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      showBoxModelHighlighterForNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const {
      node,
      hideBoxModelHighlighter,
      setSelectedNode,
      showBoxModelHighlighterForNode,
    } = this.props;

    return dom.li(
      { className: "compatibility-node-item" },
      Rep({
        defaultRep: ElementNode,
        mode: MODE.TINY,
        object: translateNodeFrontToGrip(node),
        onDOMNodeClick: () => {
          setSelectedNode(node);
          hideBoxModelHighlighter();
        },
        onDOMNodeMouseOut: () => hideBoxModelHighlighter(),
        onDOMNodeMouseOver: () => showBoxModelHighlighterForNode(node),
      })
    );
  }
}

module.exports = NodeItem;
