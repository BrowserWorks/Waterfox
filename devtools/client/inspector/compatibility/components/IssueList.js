/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const IssueItem = createFactory(
  require("devtools/client/inspector/compatibility/components/IssueItem")
);

class IssueList extends PureComponent {
  static get propTypes() {
    return {
      issues: PropTypes.arrayOf(PropTypes.shape(Types.issue)).isRequired,
      hideBoxModelHighlighter: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      showBoxModelHighlighterForNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const {
      issues,
      hideBoxModelHighlighter,
      setSelectedNode,
      showBoxModelHighlighterForNode,
    } = this.props;

    return dom.ul(
      { className: "compatibility-issue-list" },
      issues.map(issue =>
        IssueItem({
          ...issue,
          hideBoxModelHighlighter,
          setSelectedNode,
          showBoxModelHighlighterForNode,
        })
      )
    );
  }
}

module.exports = IssueList;
