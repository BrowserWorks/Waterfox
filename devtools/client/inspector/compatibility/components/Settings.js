/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const BrowserIcon = createFactory(
  require("devtools/client/inspector/compatibility/components/BrowserIcon")
);

const {
  updateSettingsVisibility,
  updateTargetBrowsers,
} = require("devtools/client/inspector/compatibility/actions/compatibility");

const CLOSE_ICON = "chrome://devtools/skin/images/close.svg";

class Settings extends PureComponent {
  static get propTypes() {
    return {
      defaultTargetBrowsers: PropTypes.arrayOf(PropTypes.shape(Types.browser))
        .isRequired,
      targetBrowsers: PropTypes.arrayOf(PropTypes.shape(Types.browser))
        .isRequired,
      updateTargetBrowsers: PropTypes.func.isRequired,
      updateSettingsVisibility: PropTypes.func.isRequired,
    };
  }

  constructor(props) {
    super(props);

    this._onTargetBrowserChanged = this._onTargetBrowserChanged.bind(this);

    this.state = {
      targetBrowsers: props.targetBrowsers,
    };
  }

  _onTargetBrowserChanged({ target }) {
    const { id, status } = target.dataset;
    let { targetBrowsers } = this.state;

    if (target.checked) {
      targetBrowsers = [...targetBrowsers, { id, status }];
    } else {
      targetBrowsers = targetBrowsers.filter(
        b => !(b.id === id && b.status === status)
      );
    }

    this.setState({ targetBrowsers });
  }

  _renderTargetBrowsers() {
    const { defaultTargetBrowsers } = this.props;
    const { targetBrowsers } = this.state;

    return dom.section(
      {
        className: "compatibility-settings__target-browsers",
      },
      dom.header(
        {
          className: "compatibility-settings__target-browsers-header",
        },
        "Target Browsers"
      ),
      dom.ul(
        {
          className: "compatibility-settings__target-browsers-list",
        },
        defaultTargetBrowsers.map(({ id, name, status, version }) => {
          const inputId = `${id}-${status}`;
          const isTargetBrowser = !!targetBrowsers.find(
            b => b.id === id && b.status === status
          );
          return dom.li(
            {
              className: "compatibility-settings__target-browsers-item",
            },
            dom.input({
              id: inputId,
              type: "checkbox",
              checked: isTargetBrowser,
              onChange: this._onTargetBrowserChanged,
              "data-id": id,
              "data-status": status,
            }),
            dom.label(
              {
                className: "compatibility-settings__target-browsers-item-label",
                htmlFor: inputId,
              },
              BrowserIcon({ id, title: `${name} ${status}` }),
              `${name} ${status} (${version})`
            )
          );
        })
      )
    );
  }

  _renderHeader() {
    return dom.header(
      {
        className: "compatibility-settings__header",
      },
      dom.label(
        {
          className: "compatibility-settings__header-label",
        },
        "Settings"
      ),
      dom.button(
        {
          className: "compatibility-settings__header-button",
          title: "Close settings",
          onClick: () => {
            const { defaultTargetBrowsers } = this.props;
            const { targetBrowsers } = this.state;

            // Sort by ordering of default browsers.
            const browsers = defaultTargetBrowsers.filter(b =>
              targetBrowsers.find(t => t.id === b.id && t.status === b.status)
            );

            if (this.props.targetBrowsers.toString() !== browsers.toString()) {
              this.props.updateTargetBrowsers(browsers);
            }

            this.props.updateSettingsVisibility();
          },
        },
        dom.img({
          className: "compatibility-settings__header-icon",
          src: CLOSE_ICON,
        })
      )
    );
  }

  render() {
    return dom.section(
      {
        className: "compatibility-settings",
      },
      this._renderHeader(),
      this._renderTargetBrowsers()
    );
  }
}

const mapStateToProps = state => {
  return {
    defaultTargetBrowsers: state.compatibility.defaultTargetBrowsers,
    targetBrowsers: state.compatibility.targetBrowsers,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateTargetBrowsers: browsers => dispatch(updateTargetBrowsers(browsers)),
    updateSettingsVisibility: () => dispatch(updateSettingsVisibility(false)),
  };
};

module.exports = connect(mapStateToProps, mapDispatchToProps)(Settings);
