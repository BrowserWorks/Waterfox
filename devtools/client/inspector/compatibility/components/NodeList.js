/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const NodeItem = createFactory(
  require("devtools/client/inspector/compatibility/components/NodeItem")
);

class NodeList extends PureComponent {
  static get propTypes() {
    return {
      nodes: PropTypes.arrayOf(Types.node).isRequired,
      hideBoxModelHighlighter: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      showBoxModelHighlighterForNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const {
      nodes,
      hideBoxModelHighlighter,
      setSelectedNode,
      showBoxModelHighlighterForNode,
    } = this.props;

    return dom.ul(
      { className: "compatibility-node-list" },
      nodes.map(node =>
        NodeItem({
          node,
          hideBoxModelHighlighter,
          setSelectedNode,
          showBoxModelHighlighterForNode,
        })
      )
    );
  }
}

module.exports = NodeList;
