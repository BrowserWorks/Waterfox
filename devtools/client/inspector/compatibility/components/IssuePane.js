/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const IssueList = createFactory(
  require("devtools/client/inspector/compatibility/components/IssueList")
);

class IssuePane extends PureComponent {
  static get propTypes() {
    return {
      issues: PropTypes.arrayOf(PropTypes.shape(Types.issue)).isRequired,
      hideBoxModelHighlighter: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      showBoxModelHighlighterForNode: PropTypes.func.isRequired,
    };
  }

  _renderNoIssues() {
    return dom.p(
      { className: "devtools-sidepanel-no-result" },
      "No compatibility issues found."
    );
  }

  render() {
    const {
      issues,
      hideBoxModelHighlighter,
      setSelectedNode,
      showBoxModelHighlighterForNode,
    } = this.props;

    return issues.length
      ? IssueList({
          issues,
          hideBoxModelHighlighter,
          setSelectedNode,
          showBoxModelHighlighterForNode,
        })
      : this._renderNoIssues();
  }
}

module.exports = IssuePane;
