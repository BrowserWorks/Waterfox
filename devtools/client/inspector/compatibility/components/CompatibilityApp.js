/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const Accordion = createFactory(
  require("devtools/client/shared/components/Accordion")
);
const Footer = createFactory(
  require("devtools/client/inspector/compatibility/components/Footer")
);
const IssuePane = createFactory(
  require("devtools/client/inspector/compatibility/components/IssuePane")
);
const Settings = createFactory(
  require("devtools/client/inspector/compatibility/components/Settings")
);

class CompatibilityApp extends PureComponent {
  static get propTypes() {
    return {
      isSettingsVisibile: PropTypes.bool.isRequired,
      isTopLevelTargetProcessing: PropTypes.bool.isRequired,
      selectedNodeIssues: PropTypes.arrayOf(PropTypes.shape(Types.issue))
        .isRequired,
      topLevelTargetIssues: PropTypes.arrayOf(PropTypes.shape(Types.issue))
        .isRequired,
      hideBoxModelHighlighter: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
      showBoxModelHighlighterForNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const {
      isSettingsVisibile,
      isTopLevelTargetProcessing,
      selectedNodeIssues,
      topLevelTargetIssues,
      hideBoxModelHighlighter,
      setSelectedNode,
      showBoxModelHighlighterForNode,
    } = this.props;

    const selectedNodeIssuePane = IssuePane({
      issues: selectedNodeIssues,
    });

    const topLevelTargetIssuePane =
      topLevelTargetIssues.length > 0 || !isTopLevelTargetProcessing
        ? IssuePane({
            issues: topLevelTargetIssues,
            hideBoxModelHighlighter,
            setSelectedNode,
            showBoxModelHighlighterForNode,
          })
        : null;

    const throbber = isTopLevelTargetProcessing
      ? dom.div({
          className: "compatibility-app__throbber devtools-throbber",
        })
      : null;

    return dom.section(
      {
        className: "compatibility-app theme-sidebar inspector-tabpanel",
      },
      dom.div(
        {
          className:
            "compatibility-app__container" +
            (isSettingsVisibile ? " compatibility-app__container-hidden" : ""),
        },
        Accordion({
          className: "compatibility-app__main",
          items: [
            {
              id: "compatibility-app--selected-element-pane",
              header: "Selected Element",
              component: selectedNodeIssuePane,
              opened: true,
            },
            {
              id: "compatibility-app--all-elements-pane",
              header: "All Issues",
              component: [topLevelTargetIssuePane, throbber],
              opened: true,
            },
          ],
        }),
        Footer({
          className: "compatibility-app__footer",
        })
      ),
      isSettingsVisibile ? Settings() : null
    );
  }
}

const mapStateToProps = state => {
  return {
    isSettingsVisibile: state.compatibility.isSettingsVisibile,
    isTopLevelTargetProcessing: state.compatibility.isTopLevelTargetProcessing,
    selectedNodeIssues: state.compatibility.selectedNodeIssues,
    topLevelTargetIssues: state.compatibility.topLevelTargetIssues,
  };
};
module.exports = connect(mapStateToProps)(CompatibilityApp);
