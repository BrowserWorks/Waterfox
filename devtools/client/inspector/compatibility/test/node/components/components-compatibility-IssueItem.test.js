/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the IssueItem component.
 */

const { shallow } = require("enzyme");
const React = require("react");

const MDNCompatibility = require("devtools/shared/compatibility/MDNCompatibility");
const IssueItem = React.createFactory(
  require("devtools/client/inspector/compatibility/components/IssueItem")
);

describe("IssueItem component", () => {
  it("renders an unsupported issue of CSS property", () => {
    const item = shallow(
      IssueItem({
        type: MDNCompatibility.ISSUE_TYPE.CSS_PROPERTY,
        property: "test-property",
        url: "test-url",
        deprecated: false,
        experimental: false,
        unsupportedBrowsers: [
          { id: "firefox", name: "Firefox", version: "70", status: "nightly" },
        ],
      })
    );
    expect(item).toMatchSnapshot();
  });

  it("renders a deprecated issue of CSS property", () => {
    const item = shallow(
      IssueItem({
        type: MDNCompatibility.ISSUE_TYPE.CSS_PROPERTY,
        property: "test-property",
        url: "test-url",
        deprecated: true,
        experimental: false,
        unsupportedBrowsers: [],
      })
    );
    expect(item).toMatchSnapshot();
  });

  it("renders an experimental issue of CSS property", () => {
    const item = shallow(
      IssueItem({
        type: MDNCompatibility.ISSUE_TYPE.CSS_PROPERTY,
        property: "test-property",
        url: "test-url",
        deprecated: false,
        experimental: true,
        unsupportedBrowsers: [],
      })
    );
    expect(item).toMatchSnapshot();
  });

  it("renders an issue which has both deprecated and experimental", () => {
    const item = shallow(
      IssueItem({
        type: MDNCompatibility.ISSUE_TYPE.CSS_PROPERTY,
        property: "test-property",
        url: "test-url",
        deprecated: true,
        experimental: true,
        unsupportedBrowsers: [],
      })
    );
    expect(item).toMatchSnapshot();
  });

  it("renders an issue which has nodes that caused this issue", () => {
    const item = shallow(
      IssueItem({
        type: MDNCompatibility.ISSUE_TYPE.CSS_PROPERTY,
        property: "test-property",
        url: "test-url",
        unsupportedBrowsers: [],
        nodes: [
          {
            actorID: "test-actor",
            attributes: [],
            nodeName: "test-element",
            nodeType: 1,
          },
        ],
      })
    );
    expect(item).toMatchSnapshot();
  });
});
