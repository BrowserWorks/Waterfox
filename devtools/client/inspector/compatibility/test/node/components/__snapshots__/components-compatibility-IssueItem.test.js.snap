// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueItem component renders a deprecated issue of CSS property 1`] = `
<li
  className="compatibility-issue-item"
  key="test-property"
>
  <div
    className="compatibility-issue-item__description compatibility-issue-item__description--deprecated"
  >
    <a
      className="compatibility-issue-item__mdn-link devtools-monospace"
      href="test-url"
      onClick={[Function]}
      title="test-url"
    >
      test-property
    </a>
    <span
      className="compatibility-issue-item__causes"
    >
      (deprecated)
    </span>
  </div>
</li>
`;

exports[`IssueItem component renders an experimental issue of CSS property 1`] = `
<li
  className="compatibility-issue-item"
  key="test-property"
>
  <div
    className="compatibility-issue-item__description compatibility-issue-item__description--experimental"
  >
    <a
      className="compatibility-issue-item__mdn-link devtools-monospace"
      href="test-url"
      onClick={[Function]}
      title="test-url"
    >
      test-property
    </a>
    <span
      className="compatibility-issue-item__causes"
    >
      (experimental)
    </span>
  </div>
</li>
`;

exports[`IssueItem component renders an issue which has both deprecated and experimental 1`] = `
<li
  className="compatibility-issue-item"
  key="test-property"
>
  <div
    className="compatibility-issue-item__description compatibility-issue-item__description--deprecated compatibility-issue-item__description--experimental"
  >
    <a
      className="compatibility-issue-item__mdn-link devtools-monospace"
      href="test-url"
      onClick={[Function]}
      title="test-url"
    >
      test-property
    </a>
    <span
      className="compatibility-issue-item__causes"
    >
      (deprecated,experimental)
    </span>
  </div>
</li>
`;

exports[`IssueItem component renders an issue which has nodes that caused this issue 1`] = `
<li
  className="compatibility-issue-item"
  key="test-property"
>
  <div
    className="compatibility-issue-item__description"
  >
    <a
      className="compatibility-issue-item__mdn-link devtools-monospace"
      href="test-url"
      onClick={[Function]}
      title="test-url"
    >
      test-property
    </a>
  </div>
  <NodePane
    nodes={
      Array [
        Object {
          "actorID": "test-actor",
          "attributes": Array [],
          "nodeName": "test-element",
          "nodeType": 1,
        },
      ]
    }
  />
</li>
`;

exports[`IssueItem component renders an unsupported issue of CSS property 1`] = `
<li
  className="compatibility-issue-item"
  key="test-property"
>
  <div
    className="compatibility-issue-item__description compatibility-issue-item__description--unsupported"
  >
    <a
      className="compatibility-issue-item__mdn-link devtools-monospace"
      href="test-url"
      onClick={[Function]}
      title="test-url"
    >
      test-property
    </a>
    <UnsupportedBrowserList
      browsers={
        Array [
          Object {
            "id": "firefox",
            "name": "Firefox",
            "status": "nightly",
            "version": "70",
          },
        ]
      }
    />
  </div>
</li>
`;
